src/SpeechCenterCredentials.cpp

```diff
@@ -9,37 +16,34 @@
 
 const char* SpeechCenterCredentials::url = "https://auth.speechcenter.verbio.com:444/api/v1/token";
 
-SpeechCenterCredentials::SpeechCenterCredentials(const std::string clientId,
-                                                 const std::string clientSecret,
-                                                 const std::string token,
-                                                 const std::string filePath)
-        : clientId{clientId},
-          clientSecret{clientSecret},
-          token{token},
-          filePath{filePath} {}
+SpeechCenterCredentials::SpeechCenterCredentials(const std::string& tokenFilePath) : tokenFilePath{tokenFilePath} {
 
-std::pair<bool, std::string> SpeechCenterCredentials::operator()() {
-
-    auto validUntil = expirationTime();
+}
 
+std::string SpeechCenterCredentials::getToken() {
+    token = readFileContent(tokenFilePath);
+    auto validUntil = decodeJwtExpirationTime();
     auto now = std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+    if (validUntil > now) return token;
 
-    if( validUntil > now)
-        return {false, ""};
-
-    auto token = newToken();
-
-    writeTokenFile(token);
-
-    return {true, token};
+    if(!clientId.empty() && !clientSecret.empty()){
+        INFO("Current token expired, requesting a new token...");
```

Comment:
i would suggest adding this  to the info in line 30:
`INFO("Current token expired or invalid, requesting a new token...");`

Reply:
done